src/ZendeskApi.Client/ZendeskApiClientFactory.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.Extensions.Options;
+using ZendeskApi.Client.Extensions;
+using ZendeskApi.Client.Options;
+
+namespace ZendeskApi.Client
+{
+    public class ZendeskApiClientFactory : IZendeskApiClient
+    {
+        private readonly ZendeskOptions _options;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public ZendeskApiClientFactory(
+            IOptions<ZendeskOptions> options,
+            IHttpClientFactory httpClientFactory)
+        {
+            _options = options.Value;
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public HttpClient CreateClient(string resource = null)
+        {
+            var formattedResource = resource?
+                .Trim('/');
```

Comment:
Can we put the `Trim` call onto the line above too?

Reply:
I've been formatting to have one operation per line as its easier to see what's happening when things change via a diff tool.